### PR TITLE
Master Sync: Fix phase seeks double-applying user tweak

### PR DESCRIPTION
**Pull request by ywwg**
Fixes lp#1879654

TESTED=rewrote broken sync test to be much more clear

### CODE REVIEW

**Review by daschuer — APPROVED** [review]
Thank you for the fix LGTM
I like to hear from @Holzhaus if his issue is fixed before merge.

**Comment by Holzhaus** [comment]
> Thank you for the fix LGTM
> I like to hear from @Holzhaus if his issue is fixed before merge.

Yup, I'll test this tomorrow. Thanks @ywwg for taking care of this!

File: src/engine/controls/bpmcontrol.h [hunk 1]
```diff
@@ -39,12 +39,16 @@ class BpmControl : public EngineControl {
     double getNearestPositionInPhase(double dThisPosition, bool respectLoops, bool playing);
     double getBeatMatchPosition(double dThisPosition, bool respectLoops, bool playing);
     double getPhaseOffset(double dThisPosition);
+    // getBeatDistance is adjusted to include the user offset so it's
+    // transparent to other decks.
```
Comment (Holzhaus) [inline src/engine/controls/bpmcontrol.h:43]:
```suggestion
    /// getBeatDistance is adjusted to include the user offset so it's
    /// transparent to other decks.
```

File: src/engine/controls/bpmcontrol.h [hunk 2]
```diff
@@ -39,12 +39,16 @@ class BpmControl : public EngineControl {
...
     double getBeatDistance(double dThisPosition) const;
 
     void setTargetBeatDistance(double beatDistance);
     void setInstantaneousBpm(double instantaneousBpm);
     void resetSyncAdjustment();
     double updateLocalBpm();
+    // updateBeatDistance is adjusted to include the user offset so
+    // it's transparent to other decks.
```
Comment (Holzhaus) [inline src/engine/controls/bpmcontrol.h:51]:
```suggestion
    /// updateBeatDistance is adjusted to include the user offset so
    /// it's transparent to other decks.
```

**Review by Holzhaus — APPROVED** [review]
Thank you very much, this works great!

**Comment by ywwg** [comment]
how are we doing backports to trunk?

**Comment by Holzhaus** [comment]
I already did.

**Comment by ywwg** [comment]
nice

**Pull request merged**